spec/controllers/orga/teams_controller_spec.rb

```diff
@@ -18,11 +18,23 @@
     include_context 'with admin logged in'
 
     describe "GET index" do
-      it "assigns all teams as @teams" do
+      let(:second_team) { create(:team, :current_season) }
+      let(:third_team) { create(:team, :current_season) }
+
+      before do
+        allow(team).to receive(:sponsored?).and_return(true)
```

Comment:
This won't stub the `sponsored?` predicate in the controller instance – it will retrieve a fresh instance from the database. The `spec/factories/teams.rb` Team factory already sets `kind` to `"sponsored"` so we currently don't have a test case for the teams that should be left out.

Proposal for updated examples (just entered it here on GH, beware of errors ;) ):

```ruby
describe 'GET index' do
  let!(:voluntary_team)  { create :team, :current_season, kind: 'voluntary' }
  let!(:sponsored_team)  { create :team, :current_season, kind: 'sponsored' }
  let!(:unaccepted_team) { create :team, :current_season, kind: nil }

  it 'assigns only selected teams as @teams' do
    get :index
    expect(assigns(:teams)).to match_array [voluntary_team, sponsored_team]
  end

  it 'assigns all teams as @teams when requested' do
    get :index, filter: 'all' # or whatever param it will turn out to be
    expect(assigns(:teams)).to match_array [voluntary_team, sponsored_team, unaccepted_team]
  end
end
```